[Jair Goh] iP

lilozz2 opened the pull request:
# New Document# Duke v.2.09

> Danger is in the eye of the storm, we must get to the centre of it


## What is in this update?

- New version releases [here](http://example.com/ "Title")
- ~~Fast~~ LIGHTSPEED response times
- ***FREE!!***

The team has been working on this for the last 2 months and we are excited to announce our new features for this quarter.

1. Improved Natural Language Understanding
We've fine-tuned the chatbot's natural language understanding capabilities. It can now better grasp context and provide more accurate responses to your queries.

2. Expanded Knowledge Base
We've enriched the chatbot's knowledge base, incorporating the latest information up to September 2021. It's now equipped with a broader range of topics, ensuring it can answer more of your questions.:blush::blush:

3. Enhanced Conversational Flow
Conversations with the chatbot will feel more natural and fluid. We've optimized the dialog management system to maintain context and provide more coherent responses.

4. New Integrations
We've added integrations with popular messaging platforms, including Slack and Microsoft Teams. You can now seamlessly deploy our chatbot within your preferred messaging environment.

5. Improved Documentation
We've revamped our documentation to make it easier for you to get started and make the most out of the chatbot's capabilities. You'll find comprehensive guides, tutorials, and examples.


---
## MyTasks
- [ ] https://github.com/octo-org/octo-repo/issues/45
- [ ] Draft task

## Sample Code

```ruby
require 'redcarpet'
markdown = Redcarpet.new("Hello World!")
puts markdown.to_html
```

Commit:
Rename, Greet & Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark as Done

Commit:
ToDo, Event, Deadline

Commit:
Automated Unit Testing

Commit:
Handling Errors

Commit:
Delete

Commit:
Save

Commit:
Merge tag 'Level-7'

Save and Load

Commit:
Dates and Times

Commit:
Merge branch 'branch-Level-8' into HEAD

Commit:
OOP

Commit:
Packages

Commit:
added gradle support

Commit:
Junit testing

Commit:
JavaDoc

Commit:
find

Commit:
Coding Standard

Commit:
Merge branch 'branch-Level-9'

Commit:
find

frrrrry commented on src/main/java/duke/Date.java:
```diff
@@ -0,0 +1,48 @@
+     * @param timeString the time of the task
+     */
+    public Date(String dateString, String timeString) {
+        String str = "2016-03-04 11:30";
```
Is there a need for the str variable?

frrrrry commented on src/main/java/duke/Date.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Override
+    public String toString() {
+        String dateString = date == null ? "" : date.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```
I like how you reduced the lines of code by using ternary operators. :+1:

frrrrry commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,168 @@
+        int taskIndex;
+        Matcher matcher;
+
+        switch(response.split(" ")[0]) {
```
Does the indentation of the switch case statements adhere to the java coding standards?

frrrrry commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,168 @@
+    public Task parseSave(String response) throws InvalidFileTypeException {
+        boolean done;
+        String[] responseList = response.split("\\|");
+        Task newtask;
```
Does the variable name newtask follow the java naming style of using the camel case syntax?

frrrrry commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,50 @@
+
+import duke.exceptions.InvalidFileTypeException;
+
+import java.io.*;
```
Is this import statement too general? Maybe can import specific classes instead :relaxed:

frrrrry commented on src/main/java/duke/command/DefaultCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+import duke.Ui;
+import duke.Storage;
+import duke.task.*;
```
Is there a need for this import statement as there are no classes from the task package used?

frrrrry commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+        return result;
+    }
+
+    public void setTaskDone(int index, boolean done) {
```
I like how you named the methods clearly as verbs. They are easy to understand :+1:

frrrrry reviewed (commented):
LGTM. Just a few nits to pick. Method naming was generally well done. But maybe can fix the coding standards violations regarding import statements and indentations for switch case statements.

iantsaii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,168 @@
+                            "Invalid use of deadline. Usage: deadline <task description> /by <date & time>"
+                    );
+                }
+                return new DeadlineCommand(matcher.group(1), matcher.group(2), false);
```
Do the statements in the case blocks follow the Single level of abstraction principle?

iantsaii reviewed (commented):
LGTM. minor nits 👍

Commit:
GUI

Commit:
GUI

Commit:
Add DeadlineCommandTest, FindCommandTest

Not enough test cases in general

A more comprehensive test bed will allow the program to continue keeping
it's function even while the code base changes and tweaks are made to it

Let's,
* Continue adding more test cases so that we can provide quality code

lilozz2 committed:
Merge pull request #2 from lilozz2/branch-A-Assertions

Branch a assertions

Commit:
Increase Readability

There was a lot of code that was uneccessary and could have been made
more concise.

Increased readability would help people understand the code better

Removed irrelevant code in Duke, Made abstraction level in Parse method
the same, changed structure of switch statement

lilozz2 committed:
Merge pull request #3 from lilozz2/branch-A-CodeQuality

Increase Readability

Commit:
C-BetterSearch

Commit:
Better GUI

Commit:
reformated some code

Commit:
add java jar file

Commit:
user guide

Commit:
More testing and changes

Commit:
bug fixes and updated user guide
